### DISCUSSION TITLE
Getting error "Could not fetch chain ID. Is your RPC URL correct?" in metamask while trying to add Anvil running on wsl

### DISCUSSION THREAD

**Discussion by roudra323** [Cyfrin/foundry-full-course-cu#1490] [Q&A] [answered]
Hey folks,
I was trying to add  Anvil on metamask, which was running in WSL.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/58447036/b7a10515-5755-4ce3-a5fd-29cbd4faea7d)

In my VScode the Anvil is running:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/58447036/dbfc7841-cdaa-43f3-918e-deaa6aa4cd51)

How to fix the issue ?

**Comment by EngrPips** [comment]
Hello @roudra323, It's very hard to say what is wrong here you would normally get that error either when anvil is not running locally or you had used https instead of http, you seem to properly set up all of those.

**Comment by Vinodnaruto** [comment]
same issue i was facing

**Reply by EngrPips** [reply]
Did you resolve it already ?

**Reply by roudra323** [reply, marked as answer]
IDK what happened but now it works.

**Comment by EvolOfThings** [comment]
Just make sure the anvil chain is running.

**Reply by EngrPips** [reply]
Indeed
